### ISSUE TITLE
mdita typed topic doesn't work

### ISSUE THREAD

**Issue by tkhm** [jelovirt/org.lwdita#33]
## Issue Summary
When I use format type "mdita", typed topic tag like `{.concept}` does not work well.

If I use it with format type "markdown", it works well, but in the README, format type "markdown" was not mentioned. So, I guess I should not use "markdown" format and should use "mdita", but I could not find how to convert these differences.

## Detail
DITA-OT version is 3.2 on Mac OS Mojave.

I have the following map.

```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">

<map>
  <title>Working in the garage</title>
  <topicref href="template-task.md" format="markdown" />
  <topicref href="template-concept.md" format="mdita" />
</map>
```

And template-task.md and template-concept.md are the followings:

template-task.md

```
---
keyword:
  - task
---

# Title of Task {.task}
Task context description.

The followings are the steps.

1. steps1

    step1 content

2. steps2

    step2 content
```

template-concept.md

```
---
keyword:
  - task
---

# Title of Concept {.concept}
Concept context description.
```

And I ran the following command:

`dita  --input="template-map.ditamap" --output="out" --format html5`

After ran it, index.html shows the following:

```
Working in the garage
    ・Title of Task
    ・Title of Concept {.concept}
    ・Title of Reference
```

As you can see, template-concept, which I set the format="mdita", shows raw characters and it looks did not work as expected. In the other hand, template-task, which I set the format="markdown", looks work as expected.

Also if I use `reltable` in map file, concept was processed "Related information", but task was processed "Related tasks".
It obviously shows format="mdita" could not process typed topic in this way.

Is there any solution?
Should I use format type "markdown" or "mdita"?

**Comment by larrykollar** [comment]
The [syntax page](https://github.com/jelovirt/org.lwdita/wiki/Syntax-reference) sort of jumbles mdita and markdown together. I've been messing with it for a while, and here's how I think it works.

BTW, the [usage page](https://github.com/jelovirt/org.lwdita/wiki/Usage) says that you can use either `markdown` or `md` to get the "loose" (I prefer to call it extended) parsing. Anyway, here are the differences as I've found them:

## mdita behavior
Specify the topic ID in the YAML header. Example:

```yaml
---
id: mytopic
---
```

CommonMark extensions in the heading such as `{.outputclass #id}` are not supported, and copied as part of the heading text.

The topic is always treated as (or converted to) a generic `<topic>` type.

Second-level headings become a nested topic.

The first paragraph always becomes a `<shortdesc>`. This becomes important if you use relative links or reltables, because the shortdesc text comes along with the topic title.

## markdown behavior
The topic ID must be specified on the heading line, using `{#topicID}`. If you put it in the YAML header, you'll get a `<data>` element in the prolog instead of the ID you wanted.

You can specify a topic type (`.concept`, `.task`, or `.reference`) along with the ID. Without a topic type, you get a generic topic. That can have minor effects in the output; for example, the default PDF plugin renders task step numbers **in bold**, where a topic's ordered list is plain.

For second-level headings, you can specify `{.section}` or `{.example}` (if the latter is allowed in your topic type). Without it, you end up with a nested topic.

What happens to the first paragraph depends on the topic type. For example, in a task, the first paragraph becomes `<context>`. In reference and (IIRC) context, it's a `<shortdesc>`.

## Bottom line
If you're composing in Markdown, planning to uplift to DITA, always use the `markdown` type in your bookmap. I've done that at work: took an old Word doc and ran it through Pandoc to get GFM, used an awk script to split it into topics (and build a bookmap), then did some cleanup and converted to DITA. (If you're doing this, use `--args.rellinks=nofamily` unless you want hard-coded parent/child/sibling links in your topics.)

If you're happy to stick with Markdown, use either `mdita` or `markdown` as the spirit moves you. And you can mix the two, depending on what you need.

**Comment by tkhm** [comment]
Thank you for your comment, @larrykollar .
There are some differences between `mdita` and `markdown` as you mentioned.
It depends what contributors and maintainers think, but I think it should be fixed......
Anyway, I'll use `markdown` for format for a while.

**Comment by jelovirt** [comment]
As said, MDITA doesn't support Markdown Extra's special attributes.
